jetty-server/src/main/java/org/eclipse/jetty/server/AsyncContentProducer.java

```diff
@@ -334,34 +334,11 @@ public boolean isReady()
 
         while (_transformedContent == null)
         {
-            if (_rawContent.isSpecial())
-            {
-                // TODO does EOF need to be passed to the interceptors?
-
-                // In case the _rawContent was set by consumeAll(), check the httpChannel
-                // to see if it has a more precise error. Otherwise, the exact same
-                // special content will be returned by the httpChannel; do not do that
-                // if the _error flag was set, meaning the current error is definitive.
-                if (!_error)
-                {
-                    HttpInput.Content refreshedRawContent = produceRawContent();
-                    if (refreshedRawContent != null)
-                        _rawContent = refreshedRawContent;
-                    _error = _rawContent.getError() != null;
-                }
-
-                if (LOG.isDebugEnabled())
-                    LOG.debug("raw content is special (with error = {}), returning it {}", _error, this);
-                return _rawContent;
-            }
-
             if (_interceptor != null)
             {
                 if (LOG.isDebugEnabled())
                     LOG.debug("using interceptor to transform raw content {}", this);
                 _transformedContent = intercept();
```

Comment:
If the interceptor returns an error or EOF, is it handled correctly in the block below?
Test cases for these 2 cases?

Reply:
Fixed and added some more tests, thanks.